src/routeGeneration/templateHelpers.ts

```diff
@@ -25,7 +25,11 @@ export function ValidateParam(property: TsoaRoute.PropertySchema, value: any, ge
       };
       return;
     } else {
-      return property.default;
+      if (property.dataType === 'any') {
+        return property.default !== undefined ? property.default : value;
```

Comment:
I think this check could be done whatever the `property.dataType` is, to prevent replacement of falsy values when `property.default` is not defined.
This was the purpose of #296

Reply:
You mean that every type should be effectively treated as nullable, not just `any` type?

Reply:
Yes I think so, because otherwise we silently replace user's input by `undefined`. 
In our case it is a problem because according to JSON API specification, we use `null` values for properties or relationships which have to be deleted in our update (PATCH) endpoints.